agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -298,13 +300,32 @@ func (state *DockerTaskEngineState) GetAllPendingEBSAttachments() []*apiresource
 func (state *DockerTaskEngineState) allPendingEBSAttachmentsUnsafe() []*apiresource.ResourceAttachment {
 	var pendingEBSAttachments []*apiresource.ResourceAttachment
 	for _, v := range state.ebsAttachments {
-		if !v.IsSent() {
+		if !v.IsAttached() {
 			pendingEBSAttachments = append(pendingEBSAttachments, v)
 		}
 	}
 	return pendingEBSAttachments
 }
 
+// GetAllPendingEBSAttachmentsWithKey returns all the ebs volumes managed by ecs on the instance that haven't been found attached on the host as a map
```

Comment:
xen-based instance does not have an volume id, would that be a concern?

Reply:
No we won't. Xen-based instances should still have a volume ID. The only issue regarding volume ID with Xen-based instances is we can't obtain it within the host instance and hence why we won't validate it for Xen-instances.

Reply:
xen based ECS EC2 instances will not advertise the requisite capability for task attach placement.